File: 141. Linked List Cycle.cpp
class Solution {
public:
    bool hasCycle(ListNode *head) {
        if (head == NULL)
            return False;
        
        ListNode *l = head;
        ListNode *p = head;
        
        while(p->next != NULL && p->next->next != NULL)
        {
            l = l->next;
            p = p->next->next;
            if(p->next == l)
            {
                break;
            }
        }
        
        if(p->next == NULL || p->next->next == NULL)
        {
            return False;
        }
        
        return True;
    }
};

File: 141. Linked List Cycle.py
#Definition for singly-linked list.
class ListNode(object):
    def __init__(self, x):
        self.val = x
        self.next = None
        
class Solution(object):
    def hasCycle(self, head):
        try:
            slow = head.next
            fast = head.next.next            
            while slow is not fast:
                slow = slow.next
                fast = fast.next.next
            return True
        except:
            return False
        
s = Solution()
has = s.hasCycle(None)
print(has)
 
 
      
      
      